src/test/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/pipeline/KendraRankingResponseProcessorTests.java

```diff
@@ -0,0 +1,112 @@
+        SearchSourceBuilder source = new SearchSourceBuilder().query(query);
+        return new SearchRequest().source(source);
+    }
+
+    private SearchResponse createResponse(int size) {
+        SearchHit[] hits = new SearchHit[size];
+        for (int i = 0; i < size; i++) {
+            Map<String, DocumentField> searchHitFields = new HashMap<>();
+            searchHitFields.put("field", new DocumentField("value" + i, Collections.emptyList()));
+            searchHitFields.put("body", new DocumentField("body" + i, Collections.emptyList()));
+            hits[i] = new SearchHit(i, "doc " + i, searchHitFields, Collections.emptyMap());
+            hits[i].sourceRef(new BytesArray("{ \"field "+ "\" : \"value"+ i + "\" ,\"body"+ "\" : \"body"+ i + "\" }"));
+            hits[i].score(i);
+        }
+        SearchHits searchHits = new SearchHits(hits, new TotalHits(size * 2L, TotalHits.Relation.EQUAL_TO), size);
+        SearchResponseSections searchResponseSections = new SearchResponseSections(searchHits, null, null, false, false, null, 0);
+        return new SearchResponse(searchResponseSections, null, 1, 1, 0, 10, null, null);
+    }
+
+    public void testFactory() throws Exception {
+
+        KendraRankingResponseProcessor.Factory factory = new KendraRankingResponseProcessor.Factory( this.clientSettings);
+
+        //test create without title field, expect exceptions
+        expectThrows(OpenSearchParseException.class, () -> factory.create(
+                Collections.emptyMap(),
+                null,
+                null,
+                Collections.emptyMap()
+        ));
+
+        //test create with required field
+        List<String> titleField= new ArrayList<>();
+        titleField.add("field");
+        Map<String,Object> configuration = new HashMap<>();
+        configuration.put("title_field","field");
+        configuration.put("body_field","body");
+        configuration.put("doc_limit","500");
+        KendraRankingResponseProcessor processorWithAllFields = factory.create(Collections.emptyMap(),"tmp0","testingAllFields", configuration);
+        assertEquals(TYPE, processorWithAllFields.getType());
+        assertEquals("tmp0", processorWithAllFields.getTag());
+        assertEquals("testingAllFields", processorWithAllFields.getDescription());
+        Map<String,Object> shortConfiguration = new HashMap<>();
+        shortConfiguration.put("body_field","body");
+        KendraRankingResponseProcessor processorWithOneFields = factory.create(Collections.emptyMap(),"tmp1","testingBodyField", shortConfiguration);
+        assertEquals(TYPE, processorWithOneFields.getType());
+        assertEquals("tmp1", processorWithOneFields.getTag());
+        assertEquals("testingBodyField", processorWithOneFields.getDescription());
+
+
+    }
+    public void testRankingResponse() throws Exception {
+        KendraHttpClient kendraClient = new KendraHttpClient(KendraClientSettings.getClientSettings(env.settings()));
```

Comment:
Does this run as a unit test? It looks like it's creating a real KendraHttpClient.

Oh... or maybe it runs but doesn't call Kendra Intelligent Ranking because the client is not valid, so the whole `isValid()` check skips it. In this case, the responses will get returned unmodified, right?

Reply:
Right. It should create a Client, but because there is not valid endpoint and credentials, so it shows warning that "it's skipping kendra client and return response without re-ranking"